### PR TITLE
Add lint and format CI steps

### LINKED ISSUE #43: Check dependencies for security vulnerabilities automatically
Audit dependencies automatically using cargo-audit as a part of the CI/CD pipeline. 
https://github.com/RustSec/cargo-audit

### LINKED ISSUE #46: Setup Clippy in Github Actions
Execute Clippy lints automatically when commits are pushed to Git:
https://github.com/marketplace/actions/rust-clippy-check

### PR DESCRIPTION
**Pull request by ollipa**
- Added Clippy linter and rustfmt format checks to CI pipeline
- Added periodic security auditing for dependencies
- Changed release pipeline to build and test in release mode

Closes #46
Closes #43

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,10 +1,9 @@
-name: Build and test
+name: ci
```
Comment (adarshk7) [inline .github/workflows/ci.yml:1]: ci -> CI

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -40,3 +39,37 @@ jobs:
...
+  format:
+    name: Run formatter
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
+      - run: rustup component add rustfmt
+      - run: cargo fmt --all -- --check
+
+  block-autosquash-commits:
```
Comment (adarshk7) [inline .github/workflows/ci.yml:66]: It doesn't matter so much but it's still nice to know if the tests passed regardless of this condition (for example, when you're still working on something)

Comment (ollipa) [reply .github/workflows/ci.yml:66]: This job is executed separately from tests so it is possible to see if tests and build pass even if this fails.

Comment (adarshk7) [reply .github/workflows/ci.yml:66]: Oh, my bad, I didn't realize

File: src/exporters/mod.rs [hunk 3]
```diff
@@ -32,12 +32,7 @@ impl ExporterScheduler {
     pub fn start(&mut self, receiver: Receiver<Message>) {
         println!("Exporter scheduler started");
 
-        loop {
-            let message = match self.receive(&receiver) {
-                Some(message) => message,
-                None => break,
-            };
-
+        while let Some(message) = self.receive(&receiver) {
```
Comment (adarshk7) [inline src/exporters/mod.rs:35]: I was just reading about this today. Very cool! 😄

**Pull request merged**